torch/_dynamo/utils.py

```diff
@@ -1589,3 +1593,37 @@ def convert(obj):
     if isinstance(___tmp_0, tuple):
         return tuple([convert(obj) for obj in ___tmp_0])
     return convert(___tmp_0)
+
+
+# NB: The dictionary has to be created lazily after TorchPatcher is called so
+# that we pick up the disabled torch.utils.checkpoint wrapper. Therefore, it is
+# sitting in a separate function.
+@functools.lru_cache(None)
+def higher_order_op_converter():
+    return {
+        torch.utils.checkpoint.checkpoint: torch._higher_order_ops.wrap.wrap_activation_checkpoint,
+    }
+
+
+def requires_higher_order_op(obj):
+    return obj in higher_order_op_converter()
+
+
+def get_higher_order_op(obj):
+    if obj is torch.utils.checkpoint.checkpoint:
+        # TODO - functionalize_rng_ops flags cannot be turned ON by default
+        # because 1) Performance concerns - seed and offset are read and passed
+        # to each AOT graph 2) Inductor has rand-specific optimizations and
+        # there is work remaining to compose them together with
+        # functionalization.
+        #
+        # Until we make it ON by default, we will have to ask users to turn on
+        # this flag manually.  TODO - Revisit if there is a simpler way to
+        # resolve this problem.
+        if not torch._functorch.config.functionalize_rng_ops:
+            raise RuntimeError(
+                "torch.compile on activation checkpointing is an experimental feature. "
+                "Please manually set torch._functorch.config.functionalize_rng_ops=True "
+                "to run torch.compile with activation checkpointing."
+            )
```

Comment:
should this be an error or should it be raise Unsupported?

I'm not sure what currently happens with torch.utils.checkpoint.checkpoint under dynamo, but if it is that it succeeds by falling back to eager-mode PyTorch, then changing it to be a RuntimeError is probably not the best UX